utilities/code-loader.go

```diff
@@ -0,0 +1,92 @@
+package utilities
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type CodeEntry struct {
+	CodeSystem string `bson:"codeSystem",json:"codeSystem"`
+	Code       string `bson:"code",json:"code"`
+	Name       string `bson:"name",json:"name"`
+}
+
+func LoadICD9FromCMS(mongoHost, lookupURL string) {
+
+	mongoSession, err := mgo.Dial(mongoHost)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	database := mongoSession.DB("fhir")
+	codeCollection := database.C("codelookup")
+
+	urlReader, err := getReaderFromUrl(lookupURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	zr, err := zip.NewReader(urlReader, int64(urlReader.Len()))
```

Comment:
I _think_ that instead of `int64(urlReader.Len())`, it should be `urlReader.Size()`.  Based on the definitions of [Len](https://golang.org/pkg/bytes/#Reader.Len) and [Size](https://golang.org/pkg/bytes/#Reader.Size), it seems more appropriate to me.  That said, since you're not doing any prior reads, they both should return the same value -- but `Size` avoids the extra conversion to int64.